[PL-Doc] Move project across organizations (#11619)
harness/developer-hub: docs/platform/organizations-and-projects/move-projects/pre-move-and-post-move-guide.md

2. Notifications:
    - If a notification rule uses a channel from the source organization, the reference will break.
    - Email notification channels using organization-level user groups will stop working.

3. Git Experience:
    - Git file paths become stale after the move, as they still reference the source organization in URLs and navigation links. 

4. Services:
    - Project-level services like manifest sources and artifact sources referencing organization-level connectors will break.

5. Environments:
    - Environment configuration files, application settings, manifest sources, or connection strings referencing organization-level resources will become unavailable.
    - Service overrides, infrastructure definitions, or GitOps clusters referencing organization-level resources will break. 

6. Monitored Services:
    - Services or environments referencing organization-level resources will become inaccessible.

7. Webhooks:
    - Git connectors, generic webhooks, or Slack webhooks using connectors or secrets referencing organization-level resources will break.
    - Custom webhook triggers will be no longer functional.

8. Access control:

>     - Organization-level RBAC policies aren’t transferred during a project move. You may need to recreate them in the destination organization to ensure proper access controls are maintained.

Reviewer: What is meant by RBAC policies?

---
title: Pre-move and post-move guide
description: Pre-move and post-move guide ensures successful project movement across organizations.
sidebar_label: Pre-move and post-move guide
sidebar_position: 2
tags:
  - project-movement
  - checklist
  - validation
keywords:
  - project move checklist
  - pre-move validation
  - post-move Remediation
---

This guide helps you decide if a project move is right for you and what to consider before finalizing the decision.

## Pre-move validation

> Before moving a project, validate the following entities to ensure they are not affected by inaccessible resources or broken references.

Reviewer: ```suggestion
Before moving a project, validate the entities to ensure they are not affected by inaccessible resources or broken references. Please note that this is not an exhaustive list.
```

4. Environments:
    - Environment configuration files, application settings, manifest sources, or connection strings referencing organization-level resources will become unavailable.
    - Service overrides, infrastructure definitions, or GitOps clusters referencing organization-level resources will break. 

5. Monitored Services:
    - Services or environments referencing organization-level resources will become inaccessible.

6. Webhooks:
    - Git connectors, generic webhooks, or Slack webhooks using connectors or secrets referencing organization-level resources will break.
    - Triggers where organization identifiers and project identifiers are present will be no longer functional.

7. Access control:
    - Organization-level RBAC policies do not transfer when a project is moved and must be reapplied in the destination organization to maintain proper access controls.
    - When a project move is initiated, all access control components are migrated asynchronously. During this process, access to the project will be temporarily blocked for users.

8. Audit logs:

>     - Historical data and execution records are fully preserved when a project is moved. However, links in the audit logs may break which redirects to the entity on the source organization.

Reviewer: ```suggestion
    - Historical audit logs are fully preserved when a project is moved. Note that the audit logs before the project was moved might contain links that refer to the older org. These links might break when accessed after the project is moved.
```

:::note Important note

> - Redirection links in pipelines, webhooks, audit logs, or other entities that reference the source organization may stop working after a project is moved, as the source organization will no longer be accessible.

Reviewer: ```suggestion
- Links in pipelines, webhooks, audit logs, or other entities that reference the older organization may stop working after a project is moved.
```

>     For example, if Project P is moved from organization O1 to O2, new audit logs will be generated under the destination organization (O2). However, existing links in the logs that reference the source organization (O1) will no longer work.

Reviewer: Why are we mentioning audit logs for the third time in the docs?

2. Notifications:
    - If a notification rule uses a channel from the source organization, the reference will break.
    - Email notification channels using organization-level user groups will stop working.

3. Services:
    - Project-level services like manifest sources and artifact sources referencing organization-level connectors will break.

4. Environments:
    - Environment configuration files, application settings, manifest sources, or connection strings referencing organization-level resources will become unavailable.
    - Service overrides, infrastructure definitions, or GitOps clusters referencing organization-level resources will break. 

5. Monitored Services:
    - Services or environments referencing organization-level resources will become inaccessible.

6. Webhooks:
    - Git connectors, generic webhooks, or Slack webhooks using connectors or secrets referencing organization-level resources will break.
    - Custom webhook triggers will be no longer functional.

7. Access control:
    - Organization-level RBAC policies do not transfer when a project is moved and and must be recreated in the destination organization to maintain proper access controls.

>     - When a project move is initiated, all project-level access control components including role bindings, resource groups, and roles are migrated asynchronously. While the move is in progress, users may experience temporary access restrictions during the move process.

Reviewer: MENTION: USERS, SERVICE ACCOUNTS, USER GROUPS too.

>     - Account-level resource groups with resource scope that includes the specificed project are moved across organizations, it will be automatically updated to reflect the new organization scope.

Author: updated with - "If an account-level resource group includes a project that is being moved, Harness automatically updates the resource group to reflect the new organization. No action is required from users."

8. Audit logs:
    - Existing audit logs remain in the source organization and are not moved with the project. Any links in these logs pointing to the moved project or older organization will become inaccessible.

:::note Important note
- Links referencing the source organization—such as those in pipelines, webhooks, or audit logs—may stop working after a project is moved. Bookmarks or URLs with account, organization, or project identifiers are not redirected and will become outdated.
- The project movement will not be allowed if a project with the same identifier already exists in the destination organization.

>     For example, if Project P is being moved from organization O1 to O2, and organization B already has a project with the same identifier (i.e., Project P), the project movement will not be allowed.

Author: Removed space
